### PR TITLE
Audio mode

### PR DESCRIPTION
**Pull request by gferraro**

### CODE REVIEW

**Review by hardiesoft — APPROVED** [review]
A few things are starting to feel like they want more documentation, but ultimately, that's kindo f a band-aid for the growing complexity of all the cases we need to handle.  I think lets get this on test.  Now that we have so many cases we probably need to re-think the flow control for tc2-agent (when we have some more time) – since it's getting pretty spaghetti, same story as the device firmware really.

File: src/main.rs [hunk 1]
```diff
@@ -877,46 +912,45 @@ fn main() {
...
                             }));
-                        }
-                        RP2040_STATE.store(state, Ordering::Relaxed);
                     }
                     TAKE_TEST_AUDIO.store(false, Ordering::Relaxed);
                 }
-
-
-                if rp2040_needs_reset {
+                if audio_mode &&  rp2040_needs_reset {
```
[the remark this answers is not in the document]
Comment (gferraro) [reply src/main.rs:936]: in thermal mode the reset logic is below, i was weary of moving it to before reading the spi data. but maybe it doens't matter?

[the remark this answers is not in the document]
Comment (hardiesoft) [reply src/main.rs:936]: That's fine then.

[the remark this answers is not in the document]
Comment (gferraro) [reply src/main.rs:936]: i'm going test moving it

**Pull request merged**